### PR TITLE
Move hello outside of the main installation

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

Move hello outside of the main installation.

## Proposal

Move the hello project to a separate example directory. 
The hello project  can be deployed by manually overriding the "templates" and "group_var" directories at the root of the project by the ones provided in this example.

### CODE REVIEW

File: group_vars/all/main.yml [hunk 1]
```diff
@@ -1,114 +1,23 @@
...
 
-# Django
-django_configuration: Development
-
-# Ports
-django_port:    8000
-cms_port:       8081
-lms_port:       8071
-richie_port:    8061
-aliases_port:   8999
```
Comment (cchaudier) [inline group_vars/all/main.yml:14]: This line must be in all for me unless if you think some project have no redirection nginx pod ?

File: group_vars/all/main.yml [hunk 2]
```diff
@@ -1,114 +1,23 @@
...
-  port: "{{ postgresql.port }}"
-  name: richie
-
 # Openshift template files
 openshift_deployments:
-  - openshift/edxapp/dc/cms.yml
-  - openshift/edxapp/dc/lms.yml
-  - openshift/edxapp/dc/memcached.yml
-  - openshift/common/dc/elasticsearch.yml
-  - openshift/common/dc/nginx.yml
```
Comment (cchaudier) [inline group_vars/all/main.yml:82]: DRY : Nginx is on all project no ?

File: group_vars/all/main.yml [hunk 3]
```diff
@@ -1,114 +1,23 @@
...
-  - openshift/richie/job/regenerate_indexes.yml
 openshift_routes:
-  - openshift/common/route/cms.yml
-  - openshift/common/route/lms.yml
-  - openshift/common/route/richie.yml
 openshift_services:
-  - openshift/common/svc/elasticsearch.yml
-  - openshift/common/svc/mongodb.yml
-  - openshift/common/svc/mysql.yml
-  - openshift/common/svc/nginx.yml
```
Comment (cchaudier) [inline group_vars/all/main.yml:103]: DRY : Nginx is on all project no ?

File: group_vars/all/openshift_routes.yml [hunk 4]
```diff
@@ -1,16 +1,14 @@
 ---
-# URL
+# Url
```
Comment (cchaudier) [inline group_vars/all/openshift_routes.yml:2]: URL is an acronym. I think it must stay in upercasse.

File: templates/openshift/hello/dc/hello.yml.j2 [hunk 5]
```diff
@@ -6,7 +6,7 @@ metadata:
   name: hello
   namespace: "{{ project_name }}"
 spec:
-  replicas: 1  # number of pods we want
+  replicas: 2  # number of pods we want
```
Comment (cchaudier) [inline templates/openshift/hello/dc/hello.yml.j2:9]: why 2 replicas ? Is for testing de blue/green switch ? the load-balancer ?

**Review by cchaudier — COMMENTED** [review]
It's the bigbang ! 
I've some comments and questions.

File: templates/openshift/hello/dc/hello.yml.j2 (hunk 5 above)
Comment (sampaccoud) [reply templates/openshift/hello/dc/hello.yml.j2:9]: yes load balancer and why not.

File: group_vars/all/main.yml (hunk 1 above)
Comment (sampaccoud) [reply group_vars/all/main.yml:14]: this comment and the following are not relevant as I am removing these files.

File: group_vars/all/openshift_routes.yml (hunk 4 above)
Comment (sampaccoud) [reply group_vars/all/openshift_routes.yml:2]: url is always mentioned in low cap in code ;-)

Comment (sampaccoud) [reply group_vars/all/openshift_routes.yml]: url is always mentioned in low cap in code ;-)

File: examples/hello/group_vars/secret/patient0/development/edxapp/credentials.vault.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (cchaudier) [inline examples/hello/group_vars/secret/patient0/development/edxapp/credentials.vault.yml]: An exemple credentials need to be vaulted ?
Are you sur that this file is linked to the hello exemple ?

**Review by cchaudier — COMMENTED** [review]
Just one question about credentials and exemple.

**Review by jmaupetit — COMMENTED** [review]
Can you document the usage of this hello example?

**Comment by sampaccoud** [comment]
@jmaupetit well, not really for the moment as it will depend on your refactoring. For the moment I was just taking it out of the main project as it has nothing to do with it.

File: examples/hello/group_vars/secret/patient0/development/richie/credentials.vault.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (cchaudier) [inline examples/hello/group_vars/secret/patient0/development/richie/credentials.vault.yml]: Are you sur that this file is linked to the hello exemple ?

**Review by cchaudier — COMMENTED** [review]
Juste some comments

**Comment by jmaupetit** [comment]
Ok, I see. I've declared an issue for that (see #46).

**Review by jmaupetit — APPROVED** [review]
Nice move.

File: examples/hello/group_vars/secret/patient0/development/edxapp/credentials.vault.yml (hunk 6 above)
Comment (sampaccoud) [reply examples/hello/group_vars/secret/patient0/development/edxapp/credentials.vault.yml]: You're right, there's no secrets files required for hello. Removed them all.

File: examples/hello/group_vars/secret/patient0/development/richie/credentials.vault.yml (hunk 7 above)
Comment (sampaccoud) [reply examples/hello/group_vars/secret/patient0/development/richie/credentials.vault.yml]: :+1: removed it

**Review by cchaudier — APPROVED** [review]

**Comment by cchaudier** [comment]
Great job !

**Pull request merged**